fedimint-server/src/consensus/server.rs

```diff
@@ -75,8 +75,8 @@ pub struct ConsensusServer {
     api_endpoints: Vec<(PeerId, SafeUrl)>,
     cfg: ServerConfig,
     submission_receiver: Receiver<ConsensusItem>,
+    shutdown_receiver: watch::Receiver<Option<u64>>,
```

Comment:
The fact that this is not persisted makes it a bit clunky in practice. Someone's fedimintd might get restarted and the whole scheduled upgrade fails because their instance ran ahead.

Reply:
Hmmm... I guess if it happened only for one member it would be fine, as other's will shut down, so maybe not a problem?

Reply:
The idea is that the index is only set a few sessions in the future and the guardians monitor the server the entire time between setting the index and the system actually shutting down to be sure it happened at the right index.

Reply:
If one member misses the session by setting the index to late the guardians would have to reorganise via chat and restart the process